cmd/storagenode/main.go

```diff
@@ -169,8 +171,15 @@ func cmdRun(cmd *cobra.Command, args []string) (err error) {
 		return err
 	}
 
+	var closeError error
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		<-c
+		closeError = peer.Close()
+	}()
+
 	runError := peer.Run(ctx)
-	closeError := peer.Close()
```

Comment:
This fails to close peer when there's no signal.

Reply:
ok, maybe one of the pieces is not canceling itself. I will take a look.